### ISSUE TITLE
Automatic switching of visibility filter in Forums

### ISSUE THREAD

**Issue by quaestor** [theCrag/website#3578]
To reproduce this, visit e.g. https://www.thecrag.com/climbing/germany/schwaebische-alb/area/249834504/forum.

Apparently, since there is a thread at exactly this level, the visibility filter is automatically set to "Descendants", which is not a real issue.

However, if you now click on "Rechter Wandbereich" in the hierarchy on the left, visibility is automatically changed to "All".

There are multiple problems with this:
1. Maybe most importantly, the semantics of the "All" filter is not entirely clear to me; the behaviour doesn't fit in the row of "Direct -- Descendants -- …"; I would expect that it shows the discussions directly at this level *and* the ones on the levels further down the tree.
2. The discussions on the current level are not separated in any way from the discussions on other levels in the tree.
3. Once a filter is chosen, it shouldn't change while browsing, only if done manually.

I guess one could fix this by
1. removing the "All" filter or clarifying its semantics (and rename it then);
2. adding a separator between "direct threads" and "descendants threads";
3. remembering the visibility filter choice while browsing.

In my opinion, this would greatly improve the usability of the forums. I often refrained from using them because I was confused about this.

**Comment by lordyavin** [comment]
"All" means all threads of ancestors (parent nodes) I think. I agree that it is confusing to see "all" threads if there is no thread at a node. Maybe  descendants should be the default in any case. Would that solve your issue?

**Comment by quaestor** [comment]
No, only this single change wouldn't fix it for me. There would still be a confusion between direct threads and descendants threads.

If "All" means ancestors, then I don't really understand what happens at "World" level.
- If set to "Direct", there are 123 discussions in "World".
- If set to "All", then there's 20 discussions, none of which is in an *ancestor* node of "World" (since there is none).
- ("Descendants" seems to work as intended, there are 3400 threads displayed at "World")

This makes no sense to me. The word "All" suggests that it should encompass *more* than "Direct" or "Descendants", clearly there's a problem here.
